### PR TITLE
feat: use SSM to manage refresh tokens [DM-1312]

### PR DESCRIPTION
**Pull request by jacobfrackson**
## Description
Instead of manually passing a valid refresh token in a config, this PR lets you create that token through the Jupyter Notebook and then use AWS' SSM to manage a chain of tokens from there. 

## Related JIRA issues
[DM-1312]

## Related changes
Upcoming Airflow PR

## Checklist

- [x] PR title follows [Commit Convention](https://www.notion.so/godialogue/Commit-Convention-84fd9a4c149e48c998d760f1c9176df0) 
- [x] PR title ends with a JIRA issue ID  
- [x] All relevant PR sections are populated, irrelevant ones are removed 
- [ ] [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests) is opened for WIP changes 
- [x] PR review requested from groups not individuals 
- [x] PR link is posted to the corresponding Slack channel 

[DM-1312]: https://dialoguemd.atlassian.net/browse/DM-1312

### CODE REVIEW

File: dia_tap_xero/credentials.py [hunk 1]
```diff
@@ -62,33 +43,18 @@ def build_oauth_headers(config):
...
     config["access_token"] = json_response["access_token"]
 
-    # Rotate token on AWS
-    # TODO add functionality to rotate the token here
+    # Rotate refresh token on AWS SSM
+    SSM.put_parameter(
+        Name="/airflow/tap_xero_refresh_token",
+        Overwrite=True,
+        Value=json_response["refresh_token"],
+        Type="SecureString",
```
Comment (mdimercurio) [inline dia_tap_xero/credentials.py:49]: doesn't the encryption/decryption require a KMS key? or it uses a default one?

File: get_initial_xero_tokens.ipynb [hunk 2]
```diff
@@ -97,11 +103,11 @@
...
+       " '950e9c71f302adedbbed6846bc7f8c1c1f4449aa2e633bcc597651c560cef752']"
```
Comment (mdimercurio) [inline get_initial_xero_tokens.ipynb]: these tokens shouldn't be version controlled.
You can remove the returned values from `XeroFirstAuth()`:
```return [json_response['access_token'], json_response['refresh_token']]
```

File: dia_tap_xero/credentials.py [hunk 3]
```diff
@@ -17,40 +17,21 @@
 # If the key does not exist: NoSuchKey
 # If not valid JSON: JSONDecodeError
 
-BASE_KEYS = ["oauth_token", "oauth_token_secret"]
-REFRESHABLE_KEYS = BASE_KEYS + ["oauth_session_handle"]
+# BASE_KEYS = ["oauth_token", "oauth_token_secret"]
+# REFRESHABLE_KEYS = BASE_KEYS + ["oauth_session_handle"]
```
Comment (mdimercurio) [inline dia_tap_xero/credentials.py]: could we remove these?

**Review by mdimercurio — CHANGES_REQUESTED** [review]
can you add a section in the readme to explain how the refresh token rotation works and which keys are being used for encryption?

File: dia_tap_xero/credentials.py (hunk 1 above)
Comment (jacobfrackson) [reply dia_tap_xero/credentials.py:49]: as long as you're in a valid AWS vault / identified that way, you don't need to pass anything else.

File: get_initial_xero_tokens.ipynb (hunk 2 above)
Comment (jacobfrackson) [reply get_initial_xero_tokens.ipynb]: Ah good catch. Yeah these tokens are no longer valid now but good catch

File: dia_tap_xero/credentials.py (hunk 3 above)
Comment (jacobfrackson) [reply dia_tap_xero/credentials.py]: already on it

**Review by mdimercurio — APPROVED** [review]

**Pull request merged**